FIRO_TSEnsembles/src/main/java/hec/SqliteDatabase.java

```diff
@@ -328,6 +341,33 @@ private EnsembleTimeSeries readEnsembleTimeSeriesFromDB(RecordIdentifier timeser
         }
         return rval;
     }
+    private EnsembleTimeSeries readEnsembleTimeSeriesFromDB(VersionIdentifier versionID, String sql) {
```

Comment:
Please consider ways to reduce the duplicate code in the two private readEnsembleTimeSeriesFromDB(..) methods.   For example the code the loops through the ResultSet could be extracted to a shared method.  Another approach could be to pass the PreparedStatement to a shared readEnsembleTimeSeriesFromDB(preparedStatement) that handles both cases for  RecordIdentifier and VersionIdentifier.

Reply:
Abstracted `readEnsembleTimeSeriesFromDb` to work a prepared statement from `RecordIdentifier` or `VersionIdentifier`